### PR TITLE
Add deregistration functionality

### PR DESCRIPTION
**Pull request by skazi0**
This is 1:1 mapping of original ruby deregistration code.

### CODE REVIEW

File: connect/client.go [hunk 1]
```diff
@@ -6,12 +6,90 @@ import (
...
+}
+
+// SMT provides one service for all products, removing it would remove all repositories.
+// Refreshing the service instead to remove the repos of deregistered product.
+func removeOrRefreshService(service Service) error {
+	if service.Name == "SMT_DUMMY_NOREMOVE_SERVICE" {
+		fmt.Println("-> Refreshing service ...")
+		refreshAllServices()
+		return nil
+	} else {
```
Comment (djoreilly) [inline connect/client.go:89]: no need for else

File: connect/client.go [hunk 2]
```diff
@@ -6,12 +6,90 @@ import (
...
+	if !CFG.Product.isEmpty() {
+		return deregisterProduct(CFG.Product)
+	} else {
+		baseProd, _ := baseProduct()
+		baseProductService, err := upgradeProduct(baseProd)
+		if err != nil {
+			return err
+		}
+
+		tree, _ := showProduct(baseProd)
```
[the remark this answers is not in the document]
Comment (skazi0) [reply connect/client.go]: done. some more error checks added as well.

File: connect/client.go [hunk 3]
```diff
@@ -6,12 +6,90 @@ import (
...
 	if !IsRegistered() {
 		return ErrSystemNotRegistered
 	}
 
-	// TODO implement deregister
-	return fmt.Errorf("Deregister not implemented yet")
+	printInformation("deregister")
+	if !CFG.Product.isEmpty() {
+		return deregisterProduct(CFG.Product)
+	} else {
```
[the remark this answers is not in the document]
Comment (skazi0) [reply connect/client.go]: done

File: connect/client.go (hunk 1 above)
Comment (skazi0) [reply connect/client.go]: done

**Review by djoreilly — APPROVED** [review]
lgtm

**Pull request merged**